test/lib/next-modes/next-dev.ts

```diff
@@ -152,7 +152,8 @@ export class NextDevInstance extends NextInstance {
 
   public override async patchFile(
     filename: string,
-    content: string | ((contents: string) => string)
+    content: string | ((contents: string) => string),
+    retryWithTempContent?: (context: { newFile: boolean }) => Promise<void>
```

Comment:
is anything using this option? I don't see a reference to it in the PR aside from adding support for it

Reply:
It's used to temporarily patch `app/default-cache/page.js` to trigger an HMR refresh:
https://github.com/vercel/next.js/pull/67925/files#diff-e8b23d459a08e352486450522b4ccaa2f103387deefa3bce3e3321e4e0092f5fR220-R229

Reply:
Do you see this being useful outside of this specific test? Wonder if we should inline it until we see a pattern

Reply:
I actually found quite a few occurrences in existing tests where the cleanup is done manually in tests. And I intended to migrate some those to this new option in a separate PR.

Reply:
Makes sense, good context for the PR description too imo. 👍

Reply:
Here's one of those follow-ups: #67965